### PR TITLE
feat(core): Admin GUI integration

### PR DESCRIPTION
**Pull request by nekolab**
Serve static assets from Kong Manager OSS

Changes include:
* new nginx config template for static file serving
* new configuration items under the `admin_gui_*` namespace
* bundle GUI assets during the build stage
* more CORS support for Admin API
* tests and more tests

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* [Implement ...]

### Issue reference
[KAG-1940](https://konghq.atlassian.net/browse/KAG-1940)

[KAG-1940]: https://konghq.atlassian.net/browse/KAG-1940?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .requirements
Comment (raoxiaoyan) [inline .requirements:18]: Maybe it names `KONG_CANOPY_VERSION`?

File: build/repositories.bzl
Comment (fffonion) [inline build/repositories.bzl:44]: let's keep this block, but just remove the invocation (Line 99) so ce to ee merge will be simpler

File: spec/fixtures/custom_nginx.template
Comment (raoxiaoyan) [inline spec/fixtures/custom_nginx.template:467]:
```suggestions
ngx.say(Kong.get_admin_gui_kconfig())
suggestions```

File: kong/admin_gui/init.lua
Comment (raoxiaoyan) [inline kong/admin_gui/init.lua:55]: Do we name to `kong:admin:gui:config`? I think we to keep the same name with `ee`. Another reason, keep the same style for the cache key. What do you think?

File: spec/02-integration/17-admin_gui/01-admin-gui-path_spec.lua
Comment (raoxiaoyan) [inline spec/02-integration/17-admin_gui/01-admin-gui-path_spec.lua:40]: I think you should remove these codes here. Because there are no `kconfig.js` to be created.

File: spec/02-integration/17-admin_gui/01-admin-gui-path_spec.lua
Comment (raoxiaoyan) [inline spec/02-integration/17-admin_gui/01-admin-gui-path_spec.lua:1]: I think you need to update this test case again. Because the retrieve `kconfig.js` changes to dynamic.

File: kong/templates/nginx_kong_gui_include.lua
Comment (raoxiaoyan) [inline kong/templates/nginx_kong_gui_include.lua:22]: Please keep simple. Like `Kong.admin_gui_config()`. Ref here: https://github.com/Kong/kong/blob/9281b4d8d57eaab16842ac95e1c0ea5f429e064b/kong/templates/nginx_kong.lua#L43.

File: kong/admin_gui/init.lua
Comment (sumimakito) [reply kong/admin_gui/init.lua:55]: What about using `admin_gui:kconfig`? 

It follows the DAO's pattern for cache keys–`<SCHEMA_NAME>:<ENTITY_ID>:...`, although we don't have a schema named `admin_gui`.

Comment (nekolab) [reply kong/admin_gui/init.lua:55]: I prefer `admin_gui:kconfig` 😄

Comment (raoxiaoyan) [reply kong/admin_gui/init.lua:55]: It's all right. Could you move the constants `K_CONFIG_CACHE_KEY` to [here](https://github.com/Kong/kong/blob/master/kong/constants.lua)?

File: build/repositories.bzl
Comment (nekolab) [reply build/repositories.bzl:99]: Yes, we don't need to add extra copyright header to GUI assets in OSS Gateway

File: kong/init.lua
Comment (raoxiaoyan) [inline kong/init.lua:1648]: `kong.cache.get` may also throw an error. I think need to handle it to get better.

Comment (raoxiaoyan) [reply kong/init.lua:1648]: Another question is when do we refresh the cache data?

Comment (sumimakito) [reply kong/init.lua:1648]: It seems that the parameters in `kconfig.js` are read and used at `kong start`, and tend to keep unchanged at runtime.

https://github.com/Kong/kong/blob/f31037a261887c1400b50d3b49ff72b4f1b0a923/kong/admin_gui/init.lua#L34-L42

It is ok, provided the current config parameters are stable at runtime. However, we'd better double-check. cc @nekolab

File: .requirements
Comment (fffonion) [inline .requirements:14]: let's align with EE, the name would be the repository name of the dependency in upper case and `-` replaced to `_`
```suggestion
KONG_ADMIN=nightly
```

**Review by fffonion — APPROVED** [review]
leaving a conditional approval: due to the size of this PR, another pair of eyes are necessary
to merge this in

File: build/repositories.bzl
Comment (oowl) [inline build/repositories.bzl:48]: It seems `kong/kong-manager-oss` is already private, This PR is still can not be built

File: kong/admin_gui/init.lua
Comment (oowl) [inline kong/admin_gui/init.lua:41]: let's use `kong.version` replace `community` to avoid CE and EE merge conflict?

**Review by oowl — APPROVED** [review]
Mostly LGTM, just need some minor adjustments.

File: build/repositories.bzl
Comment (nekolab) [reply build/repositories.bzl:48]: Yes the open source request is still in progress, we won't merge this PR until we get approval

File: kong/admin_gui/init.lua
Comment (nekolab) [reply kong/admin_gui/init.lua:41]: Have an offline chat with @oowl, we agree to unify this later

File: kong/admin_gui/init.lua
Comment (raoxiaoyan) [inline kong/admin_gui/init.lua:1]: Might create a directory `prefix`('/usr/local/kong'). Please confirm it.

**Review by oowl — APPROVED** [review]

File: kong/init.lua
Comment (windmgc) [reply kong/init.lua:1668]: Discussed offline with @nekolab and we agreed to merge this in advance to catch up with feature freeze time.

**Review by windmgc — APPROVED** [review]
Please note that this needs to be carefully cherry-picked to EE since it contains lots of minor code re-arrangements and refactors.

**Pull request merged**
